### PR TITLE
add a new test to verify reported image formats

### LINKED ISSUE #639: Required Read and Write Image Formats Untested
Currently, `check_minimum_supported` only checks for read _or_ write formats. There is no checking for read _and_ write.

Conformance should be updated to check for this.

Inspired by: https://github.com/KhronosGroup/OpenCL-CTS/pull/629#pullrequestreview-367266392

### PR DESCRIPTION
**Pull request by bashbaug**
Fixes #639 

This PR separates the checks for required image formats into a dedicated test and adds checks for required read-and-write image formats.

I think I got this right, but it would be helpful to verify that these checks are correct in the following scenarios:

1. Embedded profile implementation.
2. OpenCL 1.2 implementation, especially without 3D image write support.
3. OpenCL 3.0 implementation without read-write image support.

Thanks!

### CODE REVIEW

**Review by Jeremy-Kemp — APPROVED** [review]
Thanks for tackling this one @bashbaug 

I've tested the following configurations (all Embedded Profile) with no visible issues:

- 3.0
  - With / Without 3D image support
  - With / Without read write image support
- 1.2
  - With / Without 3D image support
  - With / Without read write image support

If another vendor could test Full Profile, that'd be great.

**Comment by bashbaug** [comment]
Waiting for one more reviewer before merging.

**Comment by lakshmih** [comment]
Qualcomm has applied this PR, re-run api conformance test and it passed.

**Review by alycm — APPROVED** [review]
LGTM.  I haven't run these tests, but two others has, so I'll merge this.

**Pull request merged**
